application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelMonitor.java

```diff
@@ -0,0 +1,245 @@
+            .filter(channelsToMonitor::containsKey)
+            .map(channelsToMonitor::get)
+            .toList();
+    }
+
+    /**
+     * Creates the status message (specific to the guild specified) that shows which channels are
+     * busy/free.
+     * <p>
+     * It gets the list of all channels in the guild and filters out all channels not being
+     * monitored (to get the correct channel order) it then updates the names of the channels in
+     * case they were changed on the guild.
+     * <p>
+     * If then iterates through all channels and checks which category they are in so that it can
+     * add the categories to the output too.
+     * 
+     * @param guild the guild the message is intended for.
+     * @return a string representing the busy/free status of channels in this guild. The String
+     *         includes emojis and other discord specific markup. Attempting to display this
+     *         somewhere other than discord will lead to unexpected results.
+     */
+    public String statusMessage(@NotNull final Guild guild) {
+        List<ChannelStatus> statusFor = guildMonitoredChannelsList(guild);
+
+        // update name so that current channel name is used
+        statusFor.forEach(channelStatus -> channelStatus.updateChannelName(guild));
+
+        // dynamically separate channels by channel categories
+        StringBuilder sb = new StringBuilder();
+        String categoryName = "";
+        for (ChannelStatus status : statusFor) {
+            Category category = guild.getGuildChannelById(status.getChannelId()).getParent();
+            if (category != null && !category.getName().equals(categoryName)) {
+                categoryName = category.getName();
+                // append the category name on a new line with markup for underlining
+                // fixme possible bug when not all channels are part of categories, may mistakenly
+                // include uncategoried channels inside previous category. will an uncategoried
+                // channel return an empty string or null? javadocs dont say.
+                sb.append("\n__").append(categoryName).append("__\n");
+            }
+            sb.append(status.toDiscord()).append("\n");
+        }
+
+        return sb.toString();
+    }
+
+    /**
+     * This method checks all channels in a guild that is currently being monitored and are
+     * currently busy and determines if the last time it was updated is more than an hour ago. If so
+     * it changes the channel's status to free.
+     * <p>
+     * This method is run automatically during startup and should be run on a 15minute schedule. The
+     * scheduled execution is not currently implemented
+     * 
+     * @param guild the guild for which to test the channel statuses of.
+     */
+    public void updateStatusFor(@NotNull Guild guild) {
+        List<ChannelStatus> statusFor = guildMonitoredChannelsList(guild);
+
+        statusFor.stream()
```

Comment:
A specific reason for not chaining this?

Reply:
hold over from before refactoring one method into many.

fixed